primitives/src/rpc/ethereum/mod.rs

```diff
@@ -5,45 +5,79 @@ mod header;
 use crate::{chain::ethereum::EthereumHeader, result::Result, rpc::RPC};
 use async_trait::async_trait;
 use reqwest::Client;
+use std::sync::atomic::{AtomicUsize, Ordering};
 
 pub use self::{block::EthBlockNumberResp, header::EthHeaderRPCResp};
 
 /// Ethereum rpc set
-pub struct EthereumRPC<'r> {
+pub struct EthereumRPC {
     /// Reqwest client
-    pub client: &'r Client,
+    pub client: Client,
     /// Rpc host
-    pub rpc: &'r str,
+    pub rpc: Vec<String>,
+    /// Counter
+    pub atom: AtomicUsize,
 }
 
-impl<'r> EthereumRPC<'r> {
+impl EthereumRPC {
     /// New EthereumRPC
-    pub fn new(client: &'r Client, rpc: &'r str) -> Self {
-        EthereumRPC { client, rpc }
+    pub fn new(client: Client, rpc: Vec<String>) -> Self {
+        EthereumRPC {
+            client,
+            rpc,
+            atom: AtomicUsize::new(0),
+        }
+    }
+
+    /// Generate random RPC
+    ///
+    /// ```
+    /// use darwinia_bridge_primitives::rpc::ethereum::EthereumRPC;
+    /// use reqwest::Client;
+    ///
+    /// let rpc = EthereumRPC::new(
+    ///   Client::new(),
+    ///   vec![
+    ///     "https://mainnet.infura.io/v3/0bfb9acbb13c426097aabb1d81a9d016".to_string(),
+    ///     "https://mainnet.infura.io/v3/74a9b1b5816b47aa853c23fcc4f2f3b6".to_string(),
+    ///   ],
+    /// );
+    ///
+    /// assert_ne!(rpc.rpc(), rpc.rpc());
+    /// ```
+    pub fn rpc(&self) -> &str {
+        self.atom.fetch_add(1, Ordering::SeqCst);
```

Comment:
Better to use self.atom.fetch_add(1, Ordering::SeqCst) % self.rpc.len() to avoid overflow?

Reply:
`self.atom.fetch_add(1, Ordering::SeqCst) % self.rpc.len() ` is in the next line